### PR TITLE
Fix several hard-dels, add harddel ref tracking

### PR DESCRIPTION
**Pull request by JohnWildkins**
tl;dr this ports updated ref tracking from tgstation/tgstation#57728 et al, and fixes a gorillion hard dels that were killing server performance and a bunch of other problems

also some miscellaneous helper stuff like text_ref() which saves some time on repeated \ref calls, and changing initatoms forloops so they're more performant

changes:
  - bugfix: "Fixed several hard-deletion issues that were causing a severe performance impact."
  - bugfix: "Fixed mobs not being removed from AI targeting spatial grid cells when they moved to a new cell."
  - bugfix: "Fixed valves not properly disconnecting, leading to a hanging hardref."
  - backend: "Added several procs to allow removing body parts from mechs, and fixed mech wreckage deleting contained parts."
  - backend: "Adjusted SSatoms init loops to be slightly more performant."
  - backend: "Changed failsafe messages to a more visible color."
  - backend: "Ported an updated version of reference tracking from tgstation; included optional debug flags to track down hardref issues."

### CODE REVIEW

**Comment by JohnWildkins** [comment]
!review

**Review by alsoandanswer — APPROVED** [review]

**Review by FluffyGhoster — APPROVED** [review]

**Review by NonQueueingMatt — APPROVED** [review]

**Review by Arrow768 — APPROVED** [review]
